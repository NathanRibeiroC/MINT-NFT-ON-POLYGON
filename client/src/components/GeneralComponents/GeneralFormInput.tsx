import "./_general_form_input.scss";
import { useState } from "react";

interface GeneralFormInputPropsType {
  labelName: string;
  id:string;
  getInputValue: (input:string, id:string) => void;
}

function GeneralFormInput(props: GeneralFormInputPropsType) {
  const [firstName, setFirstName] = useState(''); // Declare a state variable...
  return (
    <div className="gen-input-group">
      <input
        id={props.id}
        required={true}
        type="text"
        className="gen-form-input"
        onChange={e => props.getInputValue(e.target.value, props.id)}
      />
      <label className="user-label">{props.labelName}</label>
    </div>
  );
}
export default GeneralFormInput;
