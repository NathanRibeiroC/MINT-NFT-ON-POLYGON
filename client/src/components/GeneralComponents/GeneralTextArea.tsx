import './_general_text_area.scss';
import { useState } from 'react';
import isEmpty from '../../utils/isEmpty';

interface GeneralFormInputPropsType {
  labelName: string;
  id: string;
}

function GeneralTextArea({ id, labelName }: GeneralFormInputPropsType) {
  const [inputContent, setInputContent] = useState<string>('');

  return (
    <div className="gen-text-area-group">
      <textarea
        id={id}
        required
        className="gen-text-area-input"
        onBlur={(e) => setInputContent(e.target.value)}
      />
      <label htmlFor={id} className={isEmpty(inputContent) ? 'user-label-show' : 'user-label-hidde'}>{labelName}</label>
    </div>
  );
}
export default GeneralTextArea;
