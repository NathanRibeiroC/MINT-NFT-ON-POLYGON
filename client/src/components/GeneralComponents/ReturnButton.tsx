import BackArrow from '../../../public/return-arrow.svg';
import './_return_button.scss';

interface ReturnButtonPropsType {
  handleClickFunction: () => void;
  buttonType?: JSX.IntrinsicElements['button']['type'];
}

function ReturnButton({ handleClickFunction, buttonType }: ReturnButtonPropsType): JSX.Element {
  return (
    <button className="return-button" type={buttonType} onClick={handleClickFunction}>
      <img className="back-arrow-icon" alt="turn back arrow" src={BackArrow} onClick={handleClickFunction} />
    </button>
  );
}

ReturnButton.defaultProps = {
  buttonType: 'button',
};

export default ReturnButton;
