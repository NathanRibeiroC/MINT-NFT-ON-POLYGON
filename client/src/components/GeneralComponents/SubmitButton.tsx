import './_submit_button.scss';
import Buttonicon from '../../../public/contact-me-button-icon.svg';

interface SubmitButtonPropsType {
  buttonName: string;
  buttonType?: 'button' | 'submit' | 'reset' | undefined;
  handleClickFunction?: () => void;
}
function ContactMeButton({
  buttonName,
  buttonType,
  handleClickFunction,
}: SubmitButtonPropsType) {
  return (
    /* eslint-disable react/button-has-type */
    <button
      type={buttonType === undefined ? 'button' : buttonType}
      className="cssbuttons-io-button"
    >
      {buttonName}
      <div
        className="icon"
        onClick={handleClickFunction}
        role="button"
        tabIndex={0}
        onKeyDown={handleClickFunction}
      >
        <img
          className="contact-button-icon"
          src={Buttonicon}
          alt="svg-button-icon"
        />
      </div>
    </button>
    /* eslint-enable react/button-has-type */
  );
}

ContactMeButton.defaultProps = {
  buttonType: 'button',
  handleClickFunction: () => {},
};

export default ContactMeButton;
