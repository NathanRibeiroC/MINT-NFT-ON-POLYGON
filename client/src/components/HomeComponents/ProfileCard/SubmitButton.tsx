import "./_contact_me_button.scss";
import ProfileBackgroundAnimatedGradient from "./ProfileBackgroundAnimatedGradient";
import Buttonicon from "../../../../public/contact-me-button-icon.svg";
import { useAppDispatch } from "../../../redux/app/hooks";
import { flipCard } from "../../../redux/feature/flip-profile-card/flip-profile-card";

interface SubmitButtonPropsType {
  buttonName: string;
}

function ContactMeButton(props: SubmitButtonPropsType) {
  const dispatch = useAppDispatch();

  function handleClick(){
    dispatch(flipCard())
  }

  return (
    <button className="cssbuttons-io-button">
      {props.buttonName}
      <div className="icon" onClick={handleClick}>
        <img className="contact-button-icon" src={Buttonicon} alt="svg-button-icon" />
      </div>
    </button>
  );
}
export default ContactMeButton;
