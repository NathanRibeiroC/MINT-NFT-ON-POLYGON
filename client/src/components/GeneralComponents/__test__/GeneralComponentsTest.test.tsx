import { render, screen, fireEvent} from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom'
import GeneralFormInput from '../GeneralFormInput';
import GeneralTextArea from '../GeneralTextArea';
import SubmitButton from "../SubmitButton";
import ReturnButton from '../ReturnButton';

describe('GeneralFormInput.tsx', () => {

  test('GeneralFormInput.tsx render test', () => {
    const getInputValue = jest.fn(x => x);
    let tree = renderer
      .create(
        <GeneralFormInput id="fullName" labelName="Full Name" getInputValue={getInputValue}/>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('GeneralFormInput.tsx controled input test + state lift up test on getInputValue', () => {
    const getInputValue = jest.fn(x => x);
    // ARRANGE
    render(
        <GeneralFormInput id="fullName" labelName="Full Name" getInputValue={getInputValue}/>
    )
    const input = screen.getByLabelText('Full Name')
    fireEvent.change(input, {target: {value: '23'}})
    expect(getInputValue.mock.calls).toHaveLength(1);
    expect(getInputValue).toHaveBeenLastCalledWith('23','fullName');
  });
});

describe('GeneralTextArea.tsx', () => {
  test('GeneralTextArea.tsx render test', () => {
    const tree = renderer
      .create(
        <GeneralTextArea id="messageContent" labelName="Write your message here ..."/>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('GeneralTextArea.tsx placeholder disappear when writing some test', () => {
    // ARRANGE
    render(<GeneralTextArea id="messageContent" labelName="Write your message here ..."/>)
    // expect(screen.getByText("Write your message here ...")).toBeInTheDocument();
    const input = screen.getByLabelText("Write your message here ...")
    fireEvent.blur(input, { target: { value: 'Test input' } });
    // expect(input.firstChild).toHaveClass('user-label-hidde'); // class after input
    expect(input).toBeEmptyDOMElement();
  });
});

describe('SubmitButton.tsx', () => {
  test('SubmitButton.tsx render test', () => {
    const tree = renderer
      .create(
        <div>
          <SubmitButton buttonName={"TEST SUBMIT BUTTON!"}/>
        </div>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// describe('ReturnButton.tsx', () => {
//   test('SubmitButton.tsx render test', () => {
//     const tree = renderer
//       .create(
//         <div>
//           <ReturnButton/>
//         </div>
//       )
//       .toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });