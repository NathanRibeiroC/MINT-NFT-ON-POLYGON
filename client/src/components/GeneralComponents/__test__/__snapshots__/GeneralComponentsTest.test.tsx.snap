// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralFormInput.tsx GeneralFormInput.tsx render test 1`] = `
<div
  className="gen-input-group"
>
  <input
    className="gen-form-input"
    id="fullName"
    onChange={[Function]}
    required={true}
    type="text"
  />
  <label
    className="user-label"
    htmlFor="fullName"
  >
    Full Name
  </label>
</div>
`;

exports[`GeneralTextArea.tsx GeneralTextArea.tsx render test 1`] = `
<div
  className="gen-text-area-group"
>
  <textarea
    className="gen-text-area-input"
    id="messageContent"
    onBlur={[Function]}
    required={true}
  />
  <label
    className="user-label-show"
    htmlFor="messageContent"
  >
    Write your message here ...
  </label>
</div>
`;

exports[`ReturnButton.tsx SubmitButton.tsx render test 1`] = `
<div>
  <button
    className="return-button"
  >
    <img
      className="back-arrow-icon"
      src="test-file-stub"
    />
  </button>
</div>
`;

exports[`SubmitButton.tsx SubmitButton.tsx render test 1`] = `
<div>
  <button
    className="cssbuttons-io-button"
  >
    TEST SUBMIT BUTTON!
    <div
      className="icon"
    >
      <img
        alt="svg-button-icon"
        className="contact-button-icon"
        src="test-file-stub"
      />
    </div>
  </button>
</div>
`;
